cmake/modules/FindIPOPT.cmake

```diff
@@ -0,0 +1,233 @@
+        string(REPLACE "\n"                "" IPOPT_DEP ${IPOPT_DEP})
+        string(REPLACE "ipopt"             "" IPOPT_DEP ${IPOPT_DEP})       # remove any possible auto-dependency
+        separate_arguments(IPOPT_DEP)
+
+        # use the find_library command in order to prepare rpath correctly
+        foreach(LIB ${IPOPT_DEP})
+          find_library(IPOPT_SEARCH_FOR_${LIB} ${LIB} ${IPOPT_DIR}/lib
+                                                      ${IPOPT_DIR}/lib/coin
+                                                      ${IPOPT_DIR}/lib/coin/ThirdParty
+                                                      NO_DEFAULT_PATH)
+          if(IPOPT_SEARCH_FOR_${LIB})
+            # handle non-system libraries (e.g. coinblas)
+            set(IPOPT_LIBRARIES ${IPOPT_LIBRARIES} ${IPOPT_SEARCH_FOR_${LIB}})
+          else()
+            # handle system libraries (e.g. gfortran)
+            set(IPOPT_LIBRARIES ${IPOPT_LIBRARIES} ${LIB})
+          endif()
+          mark_as_advanced(IPOPT_SEARCH_FOR_${LIB})
+        endforeach()
+      endif()
+    endif()
+
+    set(IPOPT_DEFINITIONS "")
+    set(IPOPT_LINK_FLAGS "")
+  endif()
+
+# Windows platforms
+else()
+  include(SelectLibraryConfigurations)
+
+  set(IPOPT_DIR $ENV{IPOPT_DIR} CACHE PATH "Path to IPOPT build directory")
+
+  set(IPOPT_INCLUDE_DIRS ${IPOPT_DIR}/include/coin)
+  find_library(IPOPT_IPOPT_LIBRARY_RELEASE libipopt ${IPOPT_DIR}/lib
+                                                    ${IPOPT_DIR}/lib/coin
+                                                    NO_DEFAULT_PATH)
+  find_library(IPOPT_IPOPT_LIBRARY_DEBUG   libipoptD ${IPOPT_DIR}/lib
+                                                     ${IPOPT_DIR}/lib/coin
+                                                     NO_DEFAULT_PATH)
+
+  select_library_configurations(IPOPT_IPOPT)
+  set(IPOPT_LIBRARIES ${IPOPT_IPOPT_LIBRARY})
+
+  # Some old version of binary releases of IPOPT have Intel fortran
+  # libraries embedded in the library, newer releases require them to
+  # be explicitly linked.
+  if(IPOPT_IPOPT_LIBRARY)
+    get_filename_component(_MSVC_DIR "${CMAKE_LINKER}" DIRECTORY)
+
+    # Find the lib.exe executable
+    find_program(LIB_EXECUTABLE
+                 NAMES lib.exe
+                 HINTS "${_MSVC_BINDIR}"
+                       "C:/Program Files/Microsoft Visual Studio 10.0/VC/bin"
+                       "C:/Program Files (x86)/Microsoft Visual Studio 10.0/VC/bin"
+                       "C:/Program Files/Microsoft Visual Studio 11.0/VC/bin"
+                       "C:/Program Files (x86)/Microsoft Visual Studio 11.0/VC/bin"
+                       "C:/Program Files/Microsoft Visual Studio 12.0/VC/bin"
+                       "C:/Program Files (x86)/Microsoft Visual Studio 12.0/VC/bin"
+                       "C:/Program Files/Microsoft Visual Studio 14.0/VC/bin"
+                       "C:/Program Files (x86)/Microsoft Visual Studio 14.0/VC/bin"
```

Comment:
It's probably better to remove all these `HINTS` from here, as it will prefer older (possibly incompatible) versions of the executable. Also, ROOT supports only VC 15 if I'm not mistaken in any case. @bellenot Any comments here?